src/wpool_process_sup.erl

```diff
@@ -34,6 +34,7 @@ start_link(Parent, Name, Options) ->
 init({Name, Options}) ->
   Workers = proplists:get_value(workers, Options, 100),
   Strategy = proplists:get_value(strategy, Options, {one_for_one, 5, 60}),
+  maybe_add_event_handler(Options),
```

Comment:
If instead of managing a list of callbacks ourselves, we let `gen_event` take care of that, this won't be needed.

Reply:
This is still needed in order to add the initial callbacks passed in opts.

Reply:
This is still needed in order to add the initial callbacks passed in opts.